server/models/index.js

```diff
@@ -0,0 +1,54 @@
+//both: name, email, password, rating, city
+//customer: jobs: name, description, required skills, 
+//need date, city, job by id, proposals by id
+//contractor: skills, jobs completed.  Proposals: contractor
+//name, cost estimate, start date estimate, time frame estimate
+
+
+
+const baseOptions = {
+    discriminatorKey: 'userType',
+    collection: 'handymanDB',
+}
+
+//base schema: shared with both types of user
+const Base = mongoose.model('Base', new mongoose.Schema({
+    name: { 
+        type: String, 
+        required: true
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        match: [/.+@.+\..+/, 'Must match an email address!'],
+     },
+     password: {
+         type: String,
+         required: true,
+         minLength: 8,
+     },
```

Comment:
Based on how you define 'rating' here suggest a change to 'review' and then make 'rating' a float with value between 0 - 5 (or 1-5). Each review can assign a 'score' along with the text, the rating is then the average of all review scores. We can calculate the rating using a virtual type. The homework 21 assignment (book search app) has a good example of this in the User model.  

The schema for rating could be extracted to its own file and then imported and used in User. The hw 21 does this with the Book and User.